### ISSUE TITLE
[Bug]: Netdata v4.46.2 fails to build on FreeBSD 14.1 amd64

### ISSUE THREAD

**Issue by mrdaemon** [netdata/netdata#18120]
### Bug description

Hi,

I previously reported #17995 for the same issue, and it was closed as fixed with a change made as #18008.

v1.46.2 released and unfortunately I'm still encountering the exact same issue.

Full build log available here: https://gist.githubusercontent.com/mrdaemon/ad8a57b39d8984f19a389f1e33d74e00/raw/b39e7cf5f3037edb712eb1cf687f60cde7989552/netdata-build.log

Am I missing anything?

### Expected behavior

Software to compile via kickstart and/or nightly update cronjob

### Steps to reproduce

1. Run kickstart/wait for cron to fire update job on FreeBSD 14.1 on amd64
2. Software fails to build at the midpoint
3. Process stopped, netdata agent remains stopped.

### Installation method

kickstart.sh

### System info

```shell
FreeBSD okami.lab.underwares.org 14.1-RELEASE FreeBSD 14.1-RELEASE releng/14.1-n267679-10e31f0946d8 GENERIC amd64
1401000
```

### Netdata build info

```shell
# NOTE: this is the currently running version of netdata, not the one that fails to build.
Packaging:
    Netdata Version ____________________________________________ : v1.45.5
    Installation Type __________________________________________ : kickstart-build
    Package Architecture _______________________________________ : unknown
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /usr/local/etc/netdata
    Stock Configurations _______________________________________ : /usr/local/usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /usr/local/var/cache/netdata
    Permanent Databases ________________________________________ : /usr/local/var/lib/netdata
    Plugins ____________________________________________________ : /usr/local/usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/local/usr/share/netdata/web
    Log Files __________________________________________________ : /usr/local/var/log/netdata
    Lock Files _________________________________________________ : /usr/local/var/lib/netdata/lock
    Home _______________________________________________________ : /usr/local/var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : FreeBSD
    Kernel Version _____________________________________________ : 1401000
    Operating System ___________________________________________ : FreeBSD
    Operating System ID ________________________________________ : FreeBSD
    Operating System ID Like ___________________________________ : FreeBSD
    Operating System Version ___________________________________ : 14.1-RELEASE
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : uname
Hardware:
    CPU Cores __________________________________________________ :
    CPU Frequency ______________________________________________ : unknown
    RAM Bytes __________________________________________________ : 68605775872
    Disk Capacity ______________________________________________ : 139738137529344
    CPU Architecture ___________________________________________ : amd64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : sysctl
Container:
    Container __________________________________________________ : unknown
    Container Detection ________________________________________ : none
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : FreeBSD
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine ___________________________________________________ : YES
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (bundled)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : NO
    cgroup-network (associate interfaces to CGROUPS) ___________ : NO
    proc (monitor Linux systems) _______________________________ : NO
    tc (monitor Linux network QoS) _____________________________ : NO
    diskspace (monitor Linux mount points) _____________________ : NO
    freebsd (monitor FreeBSD systems) __________________________ : YES
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : NO
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : NO
    freeipmi (monitor enterprise server H/W) ___________________ : NO
    nfacct (gather netfilter accounting) _______________________ : NO
    perf (collect kernel performance events) ___________________ : NO
    slabinfo (monitor kernel object caching) ___________________ : NO
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : NO
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

### Additional info

Full build log:
https://gist.githubusercontent.com/mrdaemon/ad8a57b39d8984f19a389f1e33d74e00/raw/b39e7cf5f3037edb712eb1cf687f60cde7989552/netdata-build.log

**Comment by mrdaemon** [comment]
Tried with 4.46.3 today, it still fails in the same exact way. Happy to provide any required information, if any

**Comment by mrdaemon** [comment]
Right, so after trying a series of things, including spinning up a fresh FreeBSD 14.1 install in a vm to compare, where netdata apparently built fine, it turns out there are (at least) two issues here:

1) Even with the latest changes in #18008, the installer was still trying to use local package provided protobuf (which was pulled in by bind-tools). Removing both fixed the template warnings
2) Package abseil-20230125.3 was installed (no direct dependencies, and I sure did not install it myself, I can only surmise it was pulled in by the kickstart script in the past). Build process was definitely using it over the bundled one as well.

Removing those packages allowed the latest stable version to build.

The process of trying to clean this up was also made difficult because the uninstaller for netdata just doesn't work at the time of writing, it tries to source /etc/netdata/.environment and aborts, instead of the one in /usr/local.

If the intent is to use vendored versions of all the libraries over any lingering ones in /usr/local, I'm afraid it seems to not work as intended :(

Of course, if I can provide any information, I'd be glad to

```
$ $(ps aux | grep -m1 -E -o "[a-zA-Z/]+netdata ") -W buildinfo
Packaging:
    Netdata Version ____________________________________________ : v1.46.3
    Installation Type __________________________________________ : kickstart-build
    Package Architecture _______________________________________ : unknown
    Package Distro _____________________________________________ : unknown
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /usr/local/etc/netdata
    Stock Configurations _______________________________________ : /usr/local/usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /usr/local/var/cache/netdata
    Permanent Databases ________________________________________ : /usr/local/var/lib/netdata
    Plugins ____________________________________________________ : /usr/local/usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/local/usr/share/netdata/web
    Log Files __________________________________________________ : /usr/local/var/log/netdata
    Lock Files _________________________________________________ : /usr/local/var/lib/netdata/lock
    Home _______________________________________________________ : /usr/local/var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : FreeBSD
    Kernel Version _____________________________________________ : 1401000
    Operating System ___________________________________________ : FreeBSD
    Operating System ID ________________________________________ : FreeBSD
    Operating System ID Like ___________________________________ : FreeBSD
    Operating System Version ___________________________________ : 14.1-RELEASE
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : uname
Hardware:
    CPU Cores __________________________________________________ :
    CPU Frequency ______________________________________________ : unknown
    RAM Bytes __________________________________________________ : 68605771776
    Disk Capacity ______________________________________________ : 138751976439808
    CPU Architecture ___________________________________________ : amd64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : sysctl
Container:
    Container __________________________________________________ : unknown
    Container Detection ________________________________________ : none
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : FreeBSD
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip brotli)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : YES
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : NO
    cgroup-network (associate interfaces to CGROUPS) ___________ : NO
    proc (monitor Linux systems) _______________________________ : NO
    tc (monitor Linux network QoS) _____________________________ : NO
    diskspace (monitor Linux mount points) _____________________ : NO
    freebsd (monitor FreeBSD systems) __________________________ : YES
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : NO
    cups (monitor printers and print jobs) _____________________ : NO
    ebpf (monitor system calls) ________________________________ : NO
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : NO
    perf (collect kernel performance events) ___________________ : NO
    slabinfo (monitor kernel object caching) ___________________ : NO
    Xen ________________________________________________________ : NO
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : NO
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : NO
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

**Comment by Cris70** [comment]
Just a little update: today I was able to build Netdata 4.47.2 on Freebsd 14.0.
So this is solved for me.

**Comment by ilyam8** [comment]
@mrdaemon If you're still having the issue in v1.47.5/nightly, let us know.

**Comment by mrdaemon** [comment]
Hello, sorry about the delay, testing this was more involved than anticipated since I needed to reinstall the conflicting packages first, and the cronjob had already updated the agent.

I've reinstalled bind-tools which pulled protobuf and abseil as dependencies, and launched a rebuild of 1.47.5 via kickstart script -- I saw the configure step prepare its own protobuf and abseil patches, which seemed promising.

It is no longer using the system wide protobuf, which no longer generates the template warnings, but unfortunately it is still using the abseil headers from `/usr/local/include/abs` over the bundled one, which causes the same errors as in the previous build log.

Removing bind-tools and abseil from packages allows the build to complete without issues.

Here is [the pkg install process](https://github.com/user-attachments/files/17549397/pkg.txt) as well as the [full build log](https://github.com/user-attachments/files/17549515/buildlog.txt)
